### DISCUSSION TITLE
How can you keep classes unmangled?

### DISCUSSION THREAD

**Discussion by trungnt2910** [UlyssesWu/XamarinPosed#4] [Q&A] [answered]
Xamarin.Android seems to mangle my classes with a random meaningless CRC64 value:
![image](https://user-images.githubusercontent.com/57174311/120114975-e6c26500-c1ab-11eb-8287-5f3edc280621.png)

While your classes in `Xamarin.Posed.Demo` remains pure:
![image](https://user-images.githubusercontent.com/57174311/120114955-cdb9b400-c1ab-11eb-86e6-cd1ff112f786.png)

What did you do?

**Comment by UlyssesWu** [comment, marked as answer]
(XamarinPosed is simple, but not that simple 😉)
There might be several ways to prevent such renaming. To find such magics, you have to dig into Xamarin.Android internals. 
The method I used, is to make that class an [Android Callable Wrapper](https://docs.microsoft.com/en-us/xamarin/android/platform/java-integration/android-callable-wrappers) . ACW won't be renamed, and if you put a nested class in a ACW, it won't be renamed, either.

**Comment by trungnt2910** [comment]
I actually _did_ add the `Activity[Name="com.myname.myactivity"]`, but it seems that a rebuild is not enough, I had to clean all for the changes to apply.

Thanks.
